packages/block-editor/src/components/iframe/index.js

```diff
@@ -131,83 +135,88 @@ function Iframe( {
 	const [ containerResizeListener, { width: containerWidth } ] =
 		useResizeObserver();
 
-	const setRef = useRefEffect( ( node ) => {
-		node._load = () => {
-			setIframeDocument( node.contentDocument );
-		};
-		let iFrameDocument;
-		// Prevent the default browser action for files dropped outside of dropzones.
-		function preventFileDropDefault( event ) {
-			event.preventDefault();
-		}
-		function onLoad() {
-			const { contentDocument, ownerDocument } = node;
-			const { documentElement } = contentDocument;
-			iFrameDocument = contentDocument;
-
-			documentElement.classList.add( 'block-editor-iframe__html' );
-
-			clearerRef( documentElement );
-
-			// Ideally ALL classes that are added through get_body_class should
-			// be added in the editor too, which we'll somehow have to get from
-			// the server in the future (which will run the PHP filters).
-			setBodyClasses(
-				Array.from( ownerDocument.body.classList ).filter(
-					( name ) =>
-						name.startsWith( 'admin-color-' ) ||
-						name.startsWith( 'post-type-' ) ||
-						name === 'wp-embed-responsive'
-				)
-			);
-
-			contentDocument.dir = ownerDocument.dir;
+	const setRef = useRefEffect(
+		( node ) => {
+			node._load = () => {
+				setIframeDocument( node.contentDocument );
+			};
+			let iFrameDocument;
+			// Prevent the default browser action for files dropped outside of dropzones.
+			function preventFileDropDefault( event ) {
+				event.preventDefault();
+			}
+			function onLoad() {
+				const { contentDocument, ownerDocument } = node;
+				const { documentElement } = contentDocument;
+				iFrameDocument = contentDocument;
+
+				documentElement.classList.add( 'block-editor-iframe__html' );
+
+				clearerRef( documentElement );
+
+				// Ideally ALL classes that are added through get_body_class should
+				// be added in the editor too, which we'll somehow have to get from
+				// the server in the future (which will run the PHP filters).
+				setBodyClasses(
```

Comment:
Could you add the `rtl` body class when the site language direction is RTL?

[My attempt](https://github.com/sabernhardt/gutenberg/pull/2/files#diff-ab17abd38e687edd56573694ae3017e105e96b319571c190e9d4705c1872bc9cR160-R161) for the closely related issue #54150 was simpler but inadequate because it used `ownerDocument` (which has the profile language, not site language).

Reply:
I'll check it out, thanks for pointing it out.